DoxyFiles/man/man3/Unilan.3
.TH "Unilan" 3 "Segunda, 3 de Outubro de 2022" "Version 1.0.0" "Isotherm++" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Unilan \- Classe com as equacoes da isoterma de \fBUnilan\fP\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <Unilan\&.h>\fP
.PP
Subclasse de \fBThreeParameters\fP e \fBIsothermTemplate< Unilan >\fP\&.
.SS "Membros Públicos"

.in +1c
.ti -1c
.RI "\fBDefineIdentity\fP ('Unilan', ID::Unilan)"
.br
.RI "Definicao de \fBID\fP para esta classe para fins de identificacao de erros\&. "
.ti -1c
.RI "\fBUnilan\fP ()=default"
.br
.ti -1c
.RI "\fBUnilan\fP (const \fBUnilan\fP &_unilan)=default"
.br
.RI "Construtora de cópia\&. "
.ti -1c
.RI "virtual \fB~Unilan\fP ()=default"
.br
.RI "Destrutora\&. "
.ti -1c
.RI "\fBUnilan\fP (const Real &, const Real &, const Real &)"
.br
.RI "Construtora com os parametros que definem a isoterma de \fBUnilan\fP\&. "
.ti -1c
.RI "\fBUnilan\fP & \fBoperator=\fP (const \fBUnilan\fP &)=default"
.br
.RI "Sobrecarga do operador =\&. "
.ti -1c
.RI "Real \fBQmax\fP () const"
.br
.RI "Funcao que informa o valor da capacidade maxima de adsorcao\&. "
.ti -1c
.RI "Real \fBK1\fP () const"
.br
.RI "Funcao que retorna o coeficiente associado a constante de \fBUnilan\fP\&. "
.ti -1c
.RI "Real \fBK2\fP () const"
.br
.RI "Funcao que retorna o parametro da equacao de \fBUnilan\fP\&. "
.ti -1c
.RI "void \fBQmax\fP (const Real &_qmax)"
.br
.RI "Funcao para alterar a capacidade maxima de adsorcao\&. "
.ti -1c
.RI "void \fBK1\fP (const Real &_k1)"
.br
.RI "Funcao que altera o coeficiente associado a constante da isoterma de \fBUnilan\fP\&. "
.ti -1c
.RI "void \fBK2\fP (const Real &_k2)"
.br
.RI "Funcao para alterar o valor associado a constante da isoterma de \fBUnilan\fP\&. "
.ti -1c
.RI "Real \fBQe\fP (const Real &, const Real &) const"
.br
.RI "Funcao que calcula a quantidade de sorcao no equilibrio\&. "
.ti -1c
.RI "Real \fBQe\fP (const Real &_c) const override"
.br
.ti -1c
.RI "virtual std::unique_ptr< \fBIsotherm\fP > \fBCloneImplementation\fP () const override"
.br
.in -1c
.SS "Outros membros herdados"
.SH "Descrição detalhada"
.PP 
Classe com as equacoes da isoterma de \fBUnilan\fP\&. 

Classe com as equacoes da isoterma de \fBUnilan\fP\&.
.PP
Isoterma com tres parametros, $ q_{max} $, $ K_1 $ e $ K_2 $, cujas formulas sao as seguintes: \\begin{align} [Q_e(C_e, T)= \\frac{q_{max}}{2K_2}ln(\\frac{1+K_1C_ee^{K_2}}{1+K_1C_ee^{-K_2}})] \\end{align} Um artigo de referencia pode ser encontrado \fCaqui\fP\&. 
.PP
\fBAutores\fP
.RS 4
Joao Flavio Vieira de Vasconcellos 
.RE
.PP
\fBVersão\fP
.RS 4
1\&.0 
.RE
.PP
\fBData\fP
.RS 4
2020 
.RE
.PP
\fBBug\fP
.RS 4
Nao ha bugs conhecidos\&.
.RE
.PP
.PP
\fBCopyright\fP
.RS 4
GNU Public License\&. 
.RE
.PP

.SH "Construtores e Destrutores"
.PP 
.SS "Unilan::Unilan ()\fC [default]\fP"

.PP
Construtora default\&. 
.PP
Uso: 
.PP
.nf
Unilan  var;

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros para esta funcao 
.RE
.PP

.SS "Unilan::Unilan (const \fBUnilan\fP & _unilan)\fC [default]\fP"

.PP
Construtora de cópia\&. Uso: 
.PP
.nf
Unilan  var;
Unilan  var1(var);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_orig\fP Variavel do tipo \fBUnilan\fP original\&. 
.br
 
.RE
.PP

.SS "Unilan::Unilan (const Real & _qmax, const Real & _k1, const Real & _k2)"

.PP
Construtora com os parametros que definem a isoterma de \fBUnilan\fP\&. Uso: 
.PP
.nf
double QMAX(1\&.0);
double K1(1\&.0);    
double K2(1\&.0);        
Sips  var1(Q1, K1, K2);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_qmax\fP Capacidade maxima de adsorcao\&. 
.br
 
.br
\fI_k1\fP Constante da isortema de \fBUnilan\fP\&. 
.br
 
.br
\fI_k2\fP Constante da isoterma de \fBUnilan\fP\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_qmax\fP <= 0\&. 
.br
\fI_k1\fP <= 0\&. 
.br
\fI_k2\fP <= 0\&. 
.RE
.PP

.SH "Funções membros"
.PP 
.SS "virtual std::unique_ptr< \fBIsotherm\fP > Unilan::CloneImplementation () const\fC [inline]\fP, \fC [override]\fP, \fC [virtual]\fP"

.PP
Implementa \fBIsotherm\fP\&.
.SS "Unilan::DefineIdentity ('Unilan', ID::Unilan)"

.PP
Definicao de \fBID\fP para esta classe para fins de identificacao de erros\&. 
.SS "Real Unilan::K1 () const\fC [inline]\fP"

.PP
Funcao que retorna o coeficiente associado a constante de \fBUnilan\fP\&. Uso: 
.PP
.nf
Unilan  var1(QMAX, K1, K2);              
double k1 = var1\&.K1();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor do coeficiente associado a constante de \fBUnilan\fP\&. 
.RE
.PP

.SS "void Unilan::K1 (const Real & _k1)\fC [inline]\fP"

.PP
Funcao que altera o coeficiente associado a constante da isoterma de \fBUnilan\fP\&. Uso: 
.PP
.nf
Unilan  var1(QMAX, K1, K2);              
double k1(2\&.0);
var1\&.K1(k1);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_k1\fP Novo valor do coeficiente associado a constante da isoterma de \fBUnilan\fP\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_k1\fP <= 0\&. 
.RE
.PP

.SS "Real Unilan::K2 () const\fC [inline]\fP"

.PP
Funcao que retorna o parametro da equacao de \fBUnilan\fP\&. Uso: 
.PP
.nf
Unilan  var1(QMAX, K1, K2);              
double k2 = var1\&.K2();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor do parametro da equacao de \fBUnilan\fP\&. 
.br
 
.RE
.PP

.SS "void Unilan::K2 (const Real & _k2)\fC [inline]\fP"

.PP
Funcao para alterar o valor associado a constante da isoterma de \fBUnilan\fP\&. Uso: 
.PP
.nf
Unilan  var1(QMAX, K1, K2);              
double k2(3\&.0);
var1\&.K2(k2);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_k2\fP Novo valor associado a constante da isoterma de \fBUnilan\fP\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_k2\fP <= 0\&. 
.br
 
.RE
.PP

.SS "\fBUnilan\fP & Unilan::operator= (const \fBUnilan\fP &)\fC [default]\fP"

.PP
Sobrecarga do operador =\&. Uso: 
.PP
.nf
Unilan  var1(QMAX, K1, K2);              
Unilan  var2 = var1;

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_orig\fP Variavel do tipo \fBUnilan\fP original\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Cópia de _orig\&. 
.br
 
.RE
.PP

.SS "Real Unilan::Qe (const Real & _ce, const Real &) const\fC [virtual]\fP"

.PP
Funcao que calcula a quantidade de sorcao no equilibrio\&. Uso: 
.PP
.nf
Unilan  var1(QMAX, K1, K2);              
double ce(1\&.0);
double qe = var1\&.Qe(ce);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_c\fP Concentracao do soluto\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor da quantidade de sorcao no equilibrio\&. 
.br
 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_c\fP < 0\&. 
.br
 
.RE
.PP

.PP
Implementa \fBIsotherm\fP\&.
.SS "Real Unilan::Qe (const Real & _c) const\fC [inline]\fP, \fC [override]\fP, \fC [virtual]\fP"

.PP
Reimplementa \fBIsotherm\fP\&.
.SS "Real Unilan::Qmax () const\fC [inline]\fP"

.PP
Funcao que informa o valor da capacidade maxima de adsorcao\&. Uso: 
.PP
.nf
Unilan  var1(QMAX, K1, K2);              
double q1 = var1\&.Qmax();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor da capacidade maxima de adsorcao\&. 
.RE
.PP

.SS "void Unilan::Qmax (const Real & _qmax)\fC [inline]\fP"

.PP
Funcao para alterar a capacidade maxima de adsorcao\&. Uso: 
.PP
.nf
Unilan  var1(QMAX, K1, K2);              
double q1(3\&.0);
var1\&.Qmax(q1);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_qmax\fP Novo valor da capacidade maxima de adsorcao\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_qmax\fP <= 0\&. 
.RE
.PP


.SH "Autor"
.PP 
Gerado automaticamente por Doxygen para Isotherm++ a partir do código-fonte\&.


IsothermLib/src/Isotherm/ThreeParameters/Unilan.cpp
//==============================================================================
// Name        : Unilan.cpp
// Authors     : Lara Botelho Brum
//               Luan Rodrigues Soares de Souza
//               Joao Flavio Vieira de Vasconcellos
// Version     : 1.0
// Description : Classe com as equacoes da isoterma de Unilan.
//
// Copyright   : Copyright (C) <2022>  Joao Flavio Vasconcellos
//                                      (jflavio at iprj.uerj.br)
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//
//==============================================================================


//==============================================================================
// includes lib c++
//==============================================================================

#include <cmath>
#include <iostream>

//==============================================================================
// includes da lib IsothermLib
//==============================================================================

#include <Error/IsoException.h>
#include <Isotherm/ThreeParameters/Unilan.h>


IST_NAMESPACE_OPEN

//==============================================================================
// Variaveis estaticas
//==============================================================================

VecPairString       isothermUnilan {   PairString  ( "Qmax"
                                                    , "Capacidade maxima de adsorcao.")
                                    ,   PairString  ( "K1"
                                                    , "Constante da isoterma de Unilan.")
                                    ,   PairString  ( "K2"
                                                    , "Constante da isoterma de Unilan.")};

template<>
VecPairString IsothermTemplate < Unilan >::infoIsotherm = isothermUnilan;


//==============================================================================
// Construtora com dois parametros
//==============================================================================

#undef  __FUNCT__
#define __FUNCT__ "Unilan :: Unilan (const Real&, const Real&, const Real&)"
Unilan :: Unilan    (   const Real&     _qmax
                    ,   const Real&     _k1
                    ,   const Real&     _k2
                    ) 
                    : ThreeParameters(_qmax, _k1, _k2) {



    try {

            if (_qmax <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadQmaxLEZero
                                    );

            if (_k1 <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadK1LEZero
                                    );

            if (_k2 <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadK2LEZero
                                    );

    } catch (const IsoException& _isoExcept) {

        std::cout   << _isoExcept 
                    << "\n";
        exit(EXIT_FAILURE);

    };

    setup = true;
    
auto    val = exp(_k2);
        nume = _k1 * val;
        deno = _k1 / val;
        prod = 0.5 * _qmax / _k2;


}

//==============================================================================
// Concentracao de equilibrio Qe
//==============================================================================

#undef  __FUNCT__
#define __FUNCT__ "Unilan ::  Qe (const Real&, const Real&) const "
Real
Unilan ::  Qe   (   const Real&     _ce
                ,   const Real&
                ) const {

    try {
        if (!setup) throw
                IsoException    (   IST_LOC
                                ,   className()
                                ,   BadCoefficient);

        if (_ce < 0.0)  throw
                IsoException (IST_LOC, className(), BadCeLTZero);

    } catch (const IsoException& _isoExcept) {

        std::cout   << _isoExcept 
                    << "\n";
       exit(EXIT_FAILURE);;
    }

auto    ptrValue = std::begin(coeffValue);
auto    value = prod * log ((1 + _ce * nume) / (1 + _ce * deno));

    return (value >= ZERO ? value : 0.0)  ;
    
    
}

IST_NAMESPACE_CLOSE